cmd/branch-based-planner/main.go

```diff
@@ -46,40 +36,27 @@ func main() {
 		NewLogger(opts.logOptions).
 		WithValues("version", BuildVersion, "sha", BuildSHA)
 
-	webhookCtx, webhookCancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	pollingCtx, pollingCancel := signal.NotifyContext(context.Background(), os.Interrupt)
```

Comment:
Today I don't know why I created two of them, back than I'm sure I had reasons to do that.

Reply:
Concurrency is tricky :-) I think you can have one context, cancel from either loop. I don't think you need the latter loop to be in a goroutine, since there's nothing else you need to do. And you don't need to wait on ctx.Done() afterwards, since you'll only exit when the loop ends. (You could consider using a sync.WaitGroup if you want to make sure the goroutine has exited, before returning)

Reply:
> I don't think you need the latter loop to be in a goroutine, 

That was a request to run both of them in a separate go routine, I don't know why.